### DISCUSSION TITLE
force:source:push with sequential packages deploy resets source tracking when there is a failure in Nth package

### DISCUSSION THREAD

**Discussion by pznmc** [forcedotcom/cli#2355] [Ideas]
**Is your feature request related to a problem? Please describe.**
Failure during deployment of multiple packages using `sfdx force:source:push -f` with `pushPackageDirectoriesSequentially` set to `true` resets source tracking to state before deployment was started. When I have 50 packages in sfdx-project.json to deploy on a clean scratch org, and there is some issue in 50th package, fixing the problem and using `sfdx force:source:push -f` again causes deployment of all 50 packages, ignoring fact that the previous 49 packages have been deployed successfully. 

**What are you trying to do**
I would like to not be forced to deploy every package all over again, when there is some error in Nth package, and previous ones were deployed successfully. 

**Describe the solution you'd like**
Would be great to save state of source tracking after each successfully package deploy, instead of saving it at the end of deployment of all packages. Alternative would be some flag, that could save state after each package deployment, however I find it redundant. 

**Describe alternatives you've considered**
Didn't found any.

**Additional context**
With wider adoption of packaging, this behaviour can be annoying to more and more people, especially working on projects with many packages.

[a comment by an automated account is not shown]

**Comment by mshanemc** [comment]
we won't be making changes to `push`, it's deprecated.  See Discussion about PPDS https://github.com/forcedotcom/cli/discussions/2402

The good news is the new command let you split your pushes how you see fit (for your ex: pkgDirs 1-30, then another deploy of 30-50).  Should save a lot of time vs. PPDS when you have a large number of dirs
